Classes/Rendering/Renderer.swift

```diff
@@ -64,20 +64,30 @@ final class Renderer: Rendering {
     private let settings: CameraSettings?
     private let callbackQueue: DispatchQueue = DispatchQueue.main
     private var filter: FilterProtocol
-    private let filterFactory: FilterFactory
+    private var filterFactory: FilterFactory
     private var processingImage = false
     private var filteredPixelBuffer: CVPixelBuffer?
 
+    var filterPlatform: FilterPlatform {
+        didSet {
+            let filterFactory = FilterFactory(glContext: glContext,
+                                              metalContext: metalContext,
+                                              filterPlatform: filterPlatform)
+            self.filterFactory = filterFactory
```

Comment:
Nit: What do you think about assigning it directly to the property `filterFactory`? In this way you can also avoid to use `self`.